jacodb-core/src/main/kotlin/org/jacodb/impl/cfg/RawInstListBuilder.kt

```diff
@@ -788,26 +891,35 @@ class RawInstListBuilder(
                 it
             }
         }
+
+        fun createAssignWithNextDeclared(type: TypeName): JcRawAssignInst {
+            val assignment = nextRegisterDeclaredVariable(type, variable, insn)
+            currentFrame = currentFrame.putLocal(variable, assignment)
+            return createRawAssign(method, assignment, expr, insn)
+        }
+
         return if (oldVar != null) {
             if (oldVar is JcRawArgument) {
                 currentFrame = currentFrame.putLocal(variable, expr)
                 null
             } else if (oldVar.typeName == expr.typeName || (expr is JcRawNullConstant && !oldVar.typeName.isPrimitive)) {
                 if (!currentFrame.valueUsedInStack(oldVar) && !currentFrame.valueUsedInLocals(oldVar, variable)) {
-                    // optimization: old variable has no other usages. So we can overwrite it
-                    JcRawAssignInst(method, oldVar, expr)
+                    if (oldVar is JcRawLocalVar && infoFromLocalVars?.name == registerToLocalName[oldVar.index]) {
+                        JcRawAssignInst(method, oldVar, expr)
```

Comment:
Is it important that we don't use `createRawAssign` here?

Reply:
It's not important, although the call will not bring any new information
Put it in here anyway for the uniformity